packages/lib/services/searchengine/SearchEngine.ts

```diff
@@ -554,6 +554,94 @@ export default class SearchEngine {
 		return SearchEngine.SEARCH_TYPE_FTS;
 	}
 
+	makeSearchTree(str: string) {
+		// (tag:A (any:1 tag:B tag:C)) -> [ "tag:A", [ "any:1", "tag:B", "tag:C"]]
```

Comment:
I was going to ask how this function works, but that comment says it all. Thanks for adding it.